crates/nu-command/src/platform/input/input_.rs

```diff
@@ -59,114 +68,104 @@ impl Command for Input {
-
-        if numchar.item < 1 {
-            return Err(ShellError::UnsupportedInput {
-                msg: "Number of characters to read has to be positive".to_string(),
-                input: "value originated from here".to_string(),
-                msg_span: call.head,
-                input_span: numchar.span,
-            });
+        // Those options are not supported by reedline, default to the legacy
+        // implementation
+        let use_legacy = [
+            call.get_flag::<String>(engine_state, stack, "bytes-until-any")?
+                .is_some(),
+            call.has_flag(engine_state, stack, "suppress-output")?,
+            call.get_flag::<Spanned<i64>>(engine_state, stack, "numchar")?
+                .is_some(),
+        ]
+        .iter()
+        .any(|x| *x);
+
+        if use_legacy {
+            return self.legacy_input(engine_state, stack, call, input);
         }
 
+        let prompt_str: Option<String> = call.opt(engine_state, stack, 0)?;
         let default_val: Option<String> = call.get_flag(engine_state, stack, "default")?;
-        if let Some(prompt) = &prompt {
-            match &default_val {
-                None => print!("{prompt}"),
-                Some(val) => print!("{prompt} (default: {val})"),
+        let history_file_val: Option<String> =
+            call.get_flag(engine_state, stack, "history-file")?;
+        let max_history: usize = call
+            .get_flag::<i64>(engine_state, stack, "max-history")?
+            .map(|l| if l < 0 { 0 } else { l as usize })
+            .unwrap_or(HISTORY_SIZE);
+        let max_history_span = call.get_flag_span(stack, "max-history");
+
+        let default_str = match (&prompt_str, &default_val) {
+            (Some(_prompt), Some(val)) => format!("(default: {val}) "),
+            _ => "".to_string(),
+        };
+
+        let history_entries = match input {
+            PipelineData::Value(Value::List { vals, .. }, ..) => Some(vals),
+            _ => None,
+        };
+
+        // If we either have history entries or history file, we create an history
+        let history = match (history_entries.is_some(), history_file_val.is_some()) {
+            (false, false) => None, // Neither are set, no need for history support
+            _ => {
+                let file_history = match history_file_val {
+                    Some(file) => FileBackedHistory::with_file(max_history, file.into()),
+                    None => FileBackedHistory::new(max_history),
+                };
+                let mut history = match file_history {
+                    Ok(h) => h,
+                    Err(_e) => {
+                        return Err(ShellError::IncorrectValue {
+                            msg: "invalid max_history size".to_string(),
+                            val_span: max_history_span.expect("max-history should be set"),
+                            call_span: call.head,
+                        });
```

Comment:
Can we see based on `e` if it was the capacity or the lacking access to the directory?

https://github.com/nushell/reedline/blob/cc7960523c43dc50a383d66e745e3536afe760c7/src/history/file_backed.rs#L316-L324

Reply:
Seems to me that it only can Err [here](https://github.com/nushell/reedline/blob/cc7960523c43dc50a383d66e745e3536afe760c7/src/history/file_backed.rs#L293-297), if the capacity is max.
Am I missing something?

Reply:
In `with_file` there is a `create_dir_all` which can IO Error.

Reply:
great! let me address that!

Reply:
<img width="597" alt="image" src="https://github.com/user-attachments/assets/72a1ecc9-193d-427d-a676-99e35f03eafc" />
should be better now.